---
title: "GWAS"
author: "by [Yan Holtz](https://github.com/holtzy/) - `r format(Sys.time(), '%d %B %Y')`"
---


```{r echo=FALSE}
# Just to add space between the introduction of the document
knitr::asis_output(htmltools::htmlPreserve("<br><br>"))
```

This document describe the GWAS realised on the blood pressure data of the UKB: 

- Run the GWAS using plink
- Check result

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(qqman)
library(readr)
library(DT)
library(RColorBrewer)
```








#- Run the GWAS {.tabset .tabset-fade .tabset-pills}
***

## Data used
The GWAS is run using the following data:

- Phenotype: the phenotype file created by the script in the previous page
- Genotype: I used /afm01/Q0286/UKBiobank/v2EURu_HM3


## Plink option used


## GWAS

Go to the appropriate folder:
```{bash, eval=FALSE}
cd /home/y.holtz/BLOOD_GWAS/1_GWAS
```

Make a first try on the chromosome 1:
```{bash, eval=FALSE}
tmp_command="plink2 \
          --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr1_v2_HM3_QC\
          --allow-no-sex \
          --linear \
          --pheno /home/y.holtz/BLOOD_GWAS/0_DATA/pheno.txt.gz \
          --pheno-name f.4079.0.0 \
          --out GWAS_blood_pressure_1"
qsubshcom "$tmp_command" 1 10G test_K1 10:00:00 ""
```

Generalize on every chromosomes using an array?
```{bash, eval=FALSE}
tmp_command="plink \
          --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr{TASK_ID}_v2_HM3_QC\
          --allow-no-sex \
          --linear \
          --pheno /home/y.holtz/BLOOD_GWAS/0_DATA/pheno.txt \
          --pheno-name f.4079.0.0 \
          --out GWAS_blood_pressure_{TASK_ID}"
qsubshcom "$tmp_command" 1 30G GWAS_plink 10:00:00 "-array=1-22"
```






##Clean result 

I need to make 2 files at the end of this GWAS:

- A first output file for clumping and general description
```{bash, eval=FALSE}
# Make the file
cd /home/y.holtz/BLOOD_GWAS/1_GWAS
head -1 GWAS_blood_pressure_1.assoc.linear  > tmp
cat GWAS*linear | grep -v 'CHR'   >> tmp
# Remove weird spaces
cat tmp | awk '{ print $1,$2,$3,$4,$5,$6,$7,$8,$9 }' > result_GWAS_bloodpressure_plink.linear
gzip result_GWAS_bloodpressure_plink.linear

# Transfert it locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp  y.holtz@delta.imb.uq.edu.au:/home/y.holtz/BLOOD_GWAS/1_GWAS/result_GWAS_bloodpressure_plink.linear.gz  .
```


- A .ma file that will be used by SMR and GSMR
```{bash, eval=FALSE}
# first I need to compute the allele frequency of the reference allele with gcta (~30 minutes)
cd /home/y.holtz/BLOOD_GWAS/1_GWAS
tmp_command="gcta64 --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr{TASK_ID}_v2_HM3_QC --freq --out all_frequency_{TASK_ID}"
qsubshcom "$tmp_command" 1 30G allFreq_GCTA 10:00:00 "-array=1-22"

# Now I can build the .ma file for GSMR
echo "SNP A1 A2 freq b se p n" > GWAS_blood_pressure.ma
join <(awk '{print $2, $5, $6}' /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*bim | sort -k 1,1) <(awk '{print $2, $6}' all_frequency*.frq | sort) | join - <(zcat result_GWAS_bloodpressure_plink.linear.gz | awk '{if(NR>1) print $2, $7, $7/$8, $9, $6 }' | sort -k 1,1) >> GWAS_blood_pressure.ma
```








##Clumping
Once I have the GWAS result, I have to 'clump' it: I find all the independant pics of association, find the highest LOD, the markers involved in this pic, the position of the pic... What does it take into account? It considers: 

- LOD score of each SNP
- LD between SNPs
- Physical distance between SNPs. 

Clump -p1 is the Significance threshold for index SNPs = pic of associations. p2 is Secondary significance threshold for clumped SNPs = SNPs that are part of the association, but not pic. r2 is the LD threshold for clumping and -clump-kb is the physical distance threshold for clumping. So here I keep association with LOD score > 5 (p1: 0.00001)


```{bash, eval=FALSE}
# good folder
cd /home/y.holtz/BLOOD_GWAS/2_CLUMPING

# Prepare command
tmp_command="plink \
          --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr{TASK_ID}_v2_HM3_QC\
          --clump /home/y.holtz/BLOOD_GWAS/1_GWAS/GWAS_blood_pressure_{TASK_ID}.assoc.linear \
          --clump-p1 0.00001 --clump-p2 0.01 --clump-r2 0.50 --clump-kb 500 \
          --out /home/y.holtz/BLOOD_GWAS/2_CLUMPING/CLUMP_bloodpressure_{TASK_ID}.txt"
          
# Send qsub
qsubshcom "$tmp_command" 1 30G clumping_plink 10:00:00 "-array=1-22"
```


It is more handy to have only one file with all the chromosome. Let's build it:
```{bash, eval=FALSE}
# good folder
cd /home/y.holtz/BLOOD_GWAS/2_CLUMPING

# concatenate
cat *clumped | grep "CHR" | head -1 > result_GWAS_bloodpressure_plink_clumped
cat CLUMP*clumped | grep -v "CHR" |sed '/^$/d'  >> result_GWAS_bloodpressure_plink_clumped

# Transfert it locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp  y.holtz@delta.imb.uq.edu.au:/home/y.holtz/BLOOD_GWAS/2_CLUMPING/result_GWAS_bloodpressure_plink_clumped  .
```








#- Result {.tabset .tabset-fade .tabset-pills}
***

```{r, warning=F, message=F}
# Read result
data=read_delim("0_DATA/result_GWAS_bloodpressure_plink.linear.gz", col_names=T, delim=" ")
data=data[ sample(seq(1,nrow(data)), 10000) , ]

# Read after clumping data
clump=read.table("0_DATA/result_GWAS_bloodpressure_plink_clumped", header=T)
```

##Main statistics
```{r}
nb_snp=nrow(data)
signif = data %>% filter(P<0.00000001) %>% arrange(CHR, BP)
nb_signif=nrow(signif)
```
We had `r nb_snp` in our data set. We found `r nb_signif` significant association (with a 10^-8 threshold).  

Here is a list of all the significant assoc:
```{r}
datatable(signif, rownames = FALSE, options = list(pageLength = 5) )
```



##Clumping details
Number of unique significant loci detected after clumping: `r nrow(clump)`
Number of loci per chromosome:
```{r, warning=F, message=F}
clump %>% 
  group_by(CHR) %>%
  summarise( tot=n() ) %>%
  arrange(tot) %>%
  mutate(CHR=factor(CHR, CHR)) %>%
  ggplot( aes(x=CHR, y=tot)) +
    geom_segment(aes(x=CHR, xend=CHR, y=tot, yend=0), color="grey") +
    geom_point(size=4, color="orange") +
    coord_flip() +
    ylab("Number of significant loci") +
    xlab("Chromosome") +
    theme_bw()

```

Detail of the clumping results:
```{r, warning=F, message=F}
datatable(clump, rownames = FALSE, options = list(pageLength = 5) )
```

List of all SNP implicated in a hit?
```{r}
Hit_SNP <- c( 
  as.character(clump$SNP), 
  strsplit(as.character(clump$SP2), ',', fixed=TRUE) %>% unlist() %>% gsub("\\(.\\)", "", .) 
)
Hit_SNP <- Hit_SNP[Hit_SNP != "NONE"]
```




##Manhattan
Most common Manhattan plot
```{r, fig.width=13, fig.height=6, warning=FALSE }

manhattan(data[which(data$P<0.05) , ] , chr = 'CHR', bp = 'BP', p='P', snp = 'SNP', suggestiveline = 5, genomewideline = 8   ,highlight=Hit_SNP, annotatePval=0.0000001)

```

And with the clumping loci on it




##Pvalue distribution
```{r}
data %>% ggplot(aes(x=P)) + geom_histogram(bins=200)
```

##QQplot
```{r}
qq(data$P)
```



##Beta distribution 
Should be a normal distribution centered on 0? If Beta=0, the SNP has no effect. If absolute(Beta) is strong, SNP has a strong effect.
```{r, warning=FALSE}
data %>% ggplot(aes(x=BETA)) + geom_histogram(bins=200) +xlim(-2,2)
```


  
  
  
##Zoom
Is it possible to zoom on a specific part? Let's check the association located between 16 and 18 Mb
```{r}
data %>%
  filter(BP>16000000 & BP<18000000 & CHR==10) %>%
  ggplot( aes(x=BP/1000000, y=-log10(P))) +
    geom_point(size=1, color=alpha("skyblue", 0.6)) +
    xlab( "position on the chromosome 1 (Mbp)") +
    ylab( "LOD score") +
    ggtitle( "Loci associated with height on the chromosome 1")
```






# 9- Genes in the region 

## LocusZoom
I can use the [Locus Zoom](http://locuszoom.org/genform.php?type=yourdata) online tool to visualise the regions of interest and see the underlying genes.
Just put the .linear result of PLINK on it, and specify the SNP of interest.
It gives me this result:
 
![snp](/Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/2_TEST_ON_HEIGHT/FILES/res_rs143384.png)






## Genes in the region (IGV)
Another possiblity proposed by John would be to use the [IGV](http://software.broadinstitute.org/software/igv/) software.








## FUMA analysis

Transfert the Plink results locally:

```
cd /Users/y.holtz/Desktop
scp uqyholtz@cluster.qbi.uq.edu.au://ibscratch/wrayvisscher/Yan_Holtz/4_UKB_GWAS/2_TEST_ON_HEIGHT/1_PLINK/HAPMAP3/RESULT_GWAS/res*gz .
```

Then connect to the [FUMA](http://fuma.ctglab.nl) website. You need an account. Load this file and send the computation. Note that your file must be <600 M.
Lots of information available



  
  











