---
title: "SMR"
author: "by [Yan Holtz](https://github.com/holtzy/) - `r format(Sys.time(), '%d %B %Y')`"
---

```{r echo=FALSE}
# Just to add space between the introduction of the document
knitr::asis_output(htmltools::htmlPreserve("<br><br>"))
```

# Introduction
***
This document reproduces the example provided in the [gsmr documentation]("http://cnsgenomics.com/software/gsmr/").  
The library comes with a dataset which looks like this:

The `smr` software has been downloaded [here](http://cnsgenomics.com/software/smr/#Download) and placed in my `bin directory.

I've to specify several files to run SMR:

- **bfile**: individual-level SNP genotype data. I'm gonna use the HapMap3 data set: `/gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr` in delta.
- **gwas-summary**: the summary statistic of the GWAS I've done before. Must be in the 'ma' format. File will be: `/home/y.holtz/BLOOD_GWAS/1_GWAS/PLINK/GWAS_blood_pressure.ma`
- **beqtl-summary**: eQTL summary data coming from [Westra et al](https://www.ncbi.nlm.nih.gov/pubmed/24013639). Downloaded from [here](http://cnsgenomics.com/software/smr/#Download). Available on delta here: /home/y.holtz/BLOOD_GWAS/5_SMR/eQTL_DATA_WESTRA`
- **out**: the output name.

```{bash, eval=FALSE}
# Good directory
cd /home/y.holtz/BLOOD_GWAS/5_SMR

# Run the analysis
tmp_command="smr_Linux --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr19_v2_HM3_QC --gwas-summary /home/y.holtz/BLOOD_GWAS/1_GWAS/PLINK/GWAS_blood_pressure.ma --beqtl-summary /home/y.holtz/BLOOD_GWAS/5_SMR/eQTL_DATA_WESTRA/westra_eqtl_hg18 --out smr_BloodPressure --thread-num 1"
qsubshcom "$tmp_command" 1 30G smr_BloodPressure 10:00:00 ""
```

