<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="by Yan Holtz - 26 April 2018" />


<title>GSMR</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="site_libs/datatables-binding-0.2/datatables.js"></script>
<link href="site_libs/dt-core-1.10.12/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.12/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.12/js/jquery.dataTables.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vitamin-D GWAS in the UKB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_Pheno_Description.html">Pheno</a>
</li>
<li>
  <a href="2_GWAS.html">GWAS</a>
</li>
<li>
  <a href="6_LDSC.html">LDSC</a>
</li>
<li>
  <a href="3_SMR.html">SMR</a>
</li>
<li>
  <a href="4_GSMR.html">GSMR</a>
</li>
<li>
  <a href="7_PRS.html">PRS</a>
</li>
<li>
  <a href="8_mtCOJO.html">mtCOJO</a>
</li>
<li>
  <a href="5_XIA_et_al.html">Xia et al</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<a href="https://github.com/holtzy/" class="github-corner" aria-label="View source on Github"><svg width="80" height="80" viewBox="0 0 250 250" style="fill:#2ecc71; color:#fff; position: absolute; top: 0; border: 0; right: 0;" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">GSMR</h1>
<h4 class="author"><em>by <a href="https://github.com/holtzy/">Yan Holtz</a> - 26 April 2018</em></h4>

</div>


<p><br><br></p>
<blockquote>
<p>This document applies the <a href="%22http://cnsgenomics.com/software/gsmr/%22">gsmr tool</a> to our data: is vitamin-D causal to other traits and diseases?</p>
</blockquote>
<div id="data" class="section level1">
<h1><span class="header-section-number">1</span> - Data</h1>
<hr />
<p>To run a GSMR analysis we need:</p>
<ul>
<li>A risk factor GWAS summary statistics. <a href="https://www.nature.com/articles/s41467-017-02662-2">Xia et al.</a> recently published a GWAS on Vitamin-D, with a sample size of about 80k people. Letâ€™s load this data, and transfer it on the delta cluster. Note that a light version of this file (with p&lt;0.1) is available <a href="https://github.com/holtzy/VitaminD-GWAS/tree/master/0_DATA">here</a></li>
</ul>
<pre class="bash"><code># I download the vitamin-D summary statistics from here: https://drive.google.com/drive/folders/0BzYDtCo_doHJRFRKR0ltZHZWZjQ
# Then from local download folder to delta
gzip upload_25HydroxyVitaminD_QC.METAANALYSIS1.txt
scp upload_25HydroxyVitaminD_QC.METAANALYSIS1.txt.gz  y.holtz@delta.imb.uq.edu.au:/home/y.holtz/VITAMIND_XIA_ET_AL/1_GWAS

# good to delta folder
cd /home/y.holtz/VITAMIND_XIA_ET_AL/1_GWAS

# I miss the frequence of the major allele in this file. I can use the hapmap3 frequencies previously calculated
cat /home/y.holtz/BLOOD_GWAS/1_GWAS/PLINK/all_frequency_1.frq

# --- MA FORMAT --- make the file with this new column, transform it in a format suitable for GSMR.
echo &quot;SNP A1 A2 freq b se p n&quot; &gt; GWAS_vitaminD_XiaEtAL.ma
join &lt;(zcat upload_25HydroxyVitaminD_QC.METAANALYSIS1.txt.gz | sort -k 1,1) &lt;(awk &#39;{print $2, $6}&#39; /home/y.holtz/BLOOD_GWAS/1_GWAS/PLINK/all_frequency*.frq | sort -k 1,1) | awk &#39;{ print $1,$2,$3,$15,$4,$5,$6,$14}&#39; &gt;&gt;  GWAS_vitaminD_XiaEtAL.ma

# --- LINEAR FORMAT --- make a .linear format suitable for PLINK or FUMA
echo &quot;CHR SNP BP A1 TEST NMISS BETA STAT P&quot; &gt; GWAS_vitaminD_XiaEtAL.linear
zcat upload_25HydroxyVitaminD_QC.METAANALYSIS1.txt.gz | grep -v &#39;Chr&#39; | awk &#39;{ print $12, $1, $13, $2, &quot;ADD&quot;, $14, $4, $4*$5, $6}&#39; &gt;&gt; GWAS_vitaminD_XiaEtAL.linear
R  # We need to remove the 10^5 format for 100000 format
data=read.table(&quot;GWAS_vitaminD_XiaEtAL.linear&quot;, header=T)
data$BP = round(data$BP, 0)
options(scipen=500)
library(dplyr)
data &lt;- data %&gt;% arrange(CHR, BP)
write.table(data, file=&quot;GWAS_vitaminD_XiaEtAL.linear&quot;, quote=F, row.names=FALSE )

# --- LIGHT FORMAT --- make a light version for github
cat GWAS_vitaminD_XiaEtAL.linear | head -1 &gt;  GWAS_vitaminD_XiaEtAL.linear.light
cat GWAS_vitaminD_XiaEtAL.linear | awk &#39;{if( $9 &lt; 0.1){print $0}}&#39; &gt;&gt;  GWAS_vitaminD_XiaEtAL.linear.light
gzip GWAS_vitaminD_XiaEtAL.linear.light

# Transfer locally?
cd ~/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/home/y.holtz/VITAMIND_XIA_ET_AL/1_GWAS/GWAS_vitaminD_XiaEtAL.linear.gz .

# Note: I need to delete 2 SNPs that have a P of 0 in the R output. Because R rounds them to 0
more GWAS_vitaminD_XiaEtAL.linear | awk &#39;{ if($9!=0){print $0}}&#39; | gzip &gt; GWAS_vitaminD_XiaEtAL.linear_IGV.gz</code></pre>
<ul>
<li>GWAS summary statistics for diseases we want to explain. Iâ€™m gonna consider several sources of information. All is summarized in the Excel file called <code>list of traits GSMR</code>. We set 2 lists of outcomes. A first list with high prior. A second with all the diseases we can find. List decided by John mainly. Here is the detail of these list.</li>
</ul>
<pre class="r"><code># Reformat Parkinson&#39;s Disease data from Constanza. I need to add the allele frequency of each SNP
ls /shares/compbio/Group-Wray/YanHoltz/DATA/ALLELE_FREQUENCY/*
cd /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT 
echo &quot;SNP A1 A2 freq b se p n&quot; &gt; tmp
join &lt;(cat META_ANALYSIS_10K23_beta_se_correctGC1_pdgene_sharing_280317.tbl | grep -v &quot;^SNP&quot; | sort -k 1,1) &lt;(awk &#39;{print $2, $6}&#39; /shares/compbio/Group-Wray/YanHoltz/DATA/ALLELE_FREQUENCY/* | sort -k 1,1) | awk &#39;{ print $1,$2,$3,$9,$4,$5,$6,100000}&#39; &gt;&gt; tmp
mv tmp META_ANALYSIS_10K23_beta_se_correctGC1_pdgene_sharing_280317.ma</code></pre>
</div>
<div id="run-gsmr" class="section level1">
<h1><span class="header-section-number">2</span> - Run GSMR</h1>
<hr />
<p>The first step is to get the last GCTA version that support GSMR. Available on the QBI cluster in Zhihong folder (thanks!)</p>
<pre class="bash"><code># First transfert localy
cd /Users/y.holtz/Desktop
scp uqyholtz@inode.qbi.uq.edu.au:/ibscratch/users/uqzzhu1/code_project/gcta/build/gcta64 .

# Then from locally to delta
cd /Users/y.holtz/Desktop
scp gcta64  y.holtz@delta.imb.uq.edu.au://home/y.holtz/bin</code></pre>
<p>Then I have to prepare a couple of files and run the analysis</p>
<pre class="bash"><code># Specific repo
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR

# Prepare a file that gives the location of every b-file (one per chromosome)
ls /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*_v2_HM3_QC.bed | sed &#39;s/.bed//&#39; &gt; gsmr_ref_data.txt

# prepare a file that gives the link to the GWAS result for the risk factor = Vitamin D
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; gsmr_exposure.txt

# prepare ONE file that lists all the outcomes. This file has been build in Excel format locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp list_of_traits_GSMR.csv y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR
more list_of_traits_GSMR.csv | awk &#39;BEGIN{FS=&quot;,&quot;}{print $2, &quot;\t&quot;, $3&quot;/&quot;$4}&#39; &gt; gsmr_outcome.txt

# Split this file: one file per outcome:
split -l 1 --numeric-suffixes gsmr_outcome.txt 
for i in x* ; do a=$(echo $i | sed &#39;s/x0/x/&#39;) ; mv $i $a ; done

# send an array of GSMR
tmp_command=&quot;gcta64 --mbfile gsmr_ref_data.txt --gsmr-file gsmr_exposure.txt x{TASK_ID} --gsmr-alg 0 --out gsmr_result_vitaminDXiaEtAl_{TASK_ID}&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G GSMR_array 10:00:00 &quot;-array=1-72&quot;

# Once it&#39;s over, concatenate the results in a unique file
cat gsmr_result_vitaminDXiaEtAl_*gsmr | head -1 &gt; gsmr_result_vitaminDXiaEtAl.gsmr
cat gsmr_result_vitaminDXiaEtAl_*gsmr | grep -v &quot;Exposure&quot; &gt;&gt;  gsmr_result_vitaminDXiaEtAl.gsmr
rm *[0-9].gsmr x* *badsnps qsub*</code></pre>
<p>Then I run a reverse GSMR</p>
<pre class="r"><code># Specific repo
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR/REVERSE

# copy files done for the forward GSMR
cp ../gsmr_ref_data.txt ../gsmr_exposure.txt ../gsmr_outcome.txt .

# Split this file: one file per outcome:
split -l 1 --numeric-suffixes gsmr_outcome.txt 
for i in x* ; do a=$(echo $i | sed &#39;s/x0/x/&#39;) ; mv $i $a ; done

# send an array of GSMR
tmp_command=&quot;gcta64 --mbfile gsmr_ref_data.txt --gsmr-file gsmr_exposure.txt x{TASK_ID}  --gsmr-direction 1 --out gsmr_reverse_result_vitaminDXiaEtAl_{TASK_ID}&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G GSMR_array 10:00:00 &quot;-array=1-72&quot;

# Once it&#39;s over, concatenate the results in a unique file
cat gsmr_result_vitaminDXiaEtAl_*gsmr | head -1 &gt; gsmr_result_vitaminDXiaEtAl.gsmr
cat gsmr_result_vitaminDXiaEtAl_*gsmr | grep -v &quot;Exposure&quot; &gt;&gt;  gsmr_result_vitaminDXiaEtAl.gsmr
rm *[0-9].gsmr x* *badsnps qsub*</code></pre>
<p>First I transfer the results locally for further analysis.</p>
<pre class="bash"><code># Then from locally to delta
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR/gsmr_result_vitaminDXiaEtAl.gsmr .</code></pre>
</div>
<div id="result-primary-traits" class="section level1 tabset">
<h1><span class="header-section-number">3</span> - Result: primary traits</h1>
<hr />
<p>Description of the GSMR result:</p>
<ul>
<li>The <strong>exposure</strong> is always the Vitamin-D, and we test its causality on several outcomes.<br />
</li>
<li><strong>Bxy</strong> is the effect of the exposure on the disease. If this effect is positive, an increase of the exposure creates a increase in the Outcome.</li>
<li><strong>se</strong> is the standard error around Bxy</li>
<li><strong>p</strong> is the pvalue of the association test. We will consider a first threshold of 0.05 for significance, and a second for multiple testing correction (0.05 / # of tests)</li>
<li><strong>nsnp</strong> is the number of SNP with a significant effect on vitamin-D. It is not always the same because a few SNP are sometime discarded by the Heidi outlier test.</li>
</ul>
<pre class="r"><code># Read gsmr result
gsmr=read.table(&quot;0_DATA/gsmr_result_vitaminDXiaEtAl.gsmr&quot;, header = T)

# Read the meaning of files:
meaning=read.table(&quot;0_DATA/list_of_traits_GSMR.csv&quot;, header=T, sep=&quot;,&quot;)
nrow(meaning)</code></pre>
<pre><code>## [1] 71</code></pre>
<pre class="r"><code>primary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Main==&quot;X&quot;) ] )
secondary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Main!=&quot;X&quot;) ] )

# Compute thresholds
thres_primary &lt;- 0.05 / nrow(primary)
thres_secondary &lt;- 0.05 / nrow(secondary)</code></pre>
<p>Note that our study takes 71 traits into account. (34 primary traits and 35 traits)</p>
<div id="effect-size" class="section level2">
<h2><span class="header-section-number">3.1</span> Effect size</h2>
<p>The following chart describes the effect sizes (bxy) of each primary traits: the effect of the risk factor on the disease. A few observation:</p>
<ul>
<li>2 significant associations after multiple testing correction.</li>
<li>2 of have a positive effect size: more Vitamin-D, more trait.</li>
<li>0 have a negative effect size.</li>
</ul>
<pre class="r"><code>primary %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_primary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=name, xend=name, y=bxy-se, yend=bxy+se), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_bw() +
    theme( panel.grid.major.y = element_line(size=0.1)) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="4_GSMR_files/figure-html/unnamed-chunk-9-1.png" alt="Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size." width="768" />
<p class="caption">
Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size.
</p>
</div>
</div>
<div id="pvalues" class="section level2">
<h2><span class="header-section-number">3.2</span> Pvalues</h2>
<p>Here is an ordered barplot of the pvalues. We have several traits over the 0.05 threshold of significance. But this does not account for multiple testing.</p>
<pre class="r"><code>primary %&gt;% 
  arrange(desc(p)) %&gt;%
  filter( !is.na(bxy)) %&gt;%
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_primary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=-log10(p))) +
    geom_segment( aes(x=name, xend=name, y=-log10(p), yend=0), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_light() +
    theme( panel.grid.major.y = element_blank()) +
    ylab(&quot;-log10( pvalue )&quot;) +
    xlab(&quot;&quot;) </code></pre>
<p><img src="4_GSMR_files/figure-html/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="result-secondary-traits" class="section level1 tabset">
<h1><span class="header-section-number">4</span> - Result: secondary traits</h1>
<hr />
<div id="effect-size-1" class="section level2">
<h2><span class="header-section-number">4.1</span> Effect size</h2>
<p>A few observation:</p>
<ul>
<li>3 significant associations after multiple testing correction.</li>
<li>3 of have a positive effect size: more Vitamin-D, more trait.</li>
<li>0 have a negative effect size.</li>
</ul>
<pre class="r"><code>secondary %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_secondary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=name, xend=name, y=bxy-se, yend=bxy+se), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_bw() +
    theme( panel.grid.major.y = element_line(size=0.1)) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="4_GSMR_files/figure-html/unnamed-chunk-11-1.png" alt="Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size." width="768" />
<p class="caption">
Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size.
</p>
</div>
</div>
<div id="pvalues-1" class="section level2">
<h2><span class="header-section-number">4.2</span> Pvalues</h2>
<p>Here is an ordered barplot of the pvalues. We have several traits over the 0.05 threshold of significance. But this does not account for multiple testing.</p>
<pre class="r"><code>secondary %&gt;% 
  arrange(desc(p)) %&gt;%
  filter( !is.na(bxy)) %&gt;%
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_secondary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=-log10(p))) +
    geom_segment( aes(x=name, xend=name, y=-log10(p), yend=0), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_light() +
    theme( panel.grid.major.y = element_blank()) +
    ylab(&quot;-log10( pvalue )&quot;) +
    xlab(&quot;&quot;) </code></pre>
<p><img src="4_GSMR_files/figure-html/unnamed-chunk-12-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="table" class="section level1">
<h1><span class="header-section-number">5</span> - Table</h1>
<hr />
<p>The 71 GSMR test results are displayed in this table in case you need to check a specific value:</p>
<pre class="r"><code>datatable(gsmr %&gt;% arrange(p), rownames = FALSE, options = list(pageLength = 15) )</code></pre>
<div id="htmlwidget-c50503bd870f8fe78ee6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c50503bd870f8fe78ee6">{"x":{"filter":"none","data":[["vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD"],["Myeloid","Granulocytecount","Whitebloodcellcount","Neutrophil(x4)","Height","Alzheimer'sDisease","Basophilcount","PlateletCount","Lymphocytecount","HDLcholesterol","allergicrhinitis","Plateletcrit","Monocytecount","Dementia-Alzheimer","hypertensivedisease","insomnia","MeanCorpuscularHemoglobin","maculardegeneration","osteoarthritis","irondeficiencyanemias","systolicbloodpressure","acutereactiontostress","Reticulocytecount","Plateletditributionwidth","Eosinophilicesophagitis","IQ(UKB)","herniaabdominopelviccavity","BMI","DepressiveSymptoms","UlcerativeColitis","psychiatricdisorder","majordepressivedisorder","CAD","Hemoglobin","AutismSpectrumDisorder","dermatophytosis","hemorrhoids","MajorDepressiveDisorder","Redcelldistributionwidth","dyslipidemia","typeIIdiabetes","peripheralvasculardisease","Meanplateletvolume","Neuroticism","diastolicbloodpressure","IQ","Schizophrenia","pepticulcers","RheumatoidArthritis","Parkinson","AgedMacularDegeneration","cardiovasculardisease","irritablebowelsyndrome","Waist-to-hipratioadjustedforBMI","varicoseveinsoflowerextremities","asthma","Crohn'sDisease","BipolarDisorder","diseasecount","Hematocrit","LDLcholesterol","ICV","Wellbeing(x2)","Redbloodcellcount","Meancorpuscularvolume","osteoporosis","EduYear","triglycerides","cancer","ImmatureFractionofReticulocytes","typeIIdiabetes"],[0.188634,0.183935,0.184679,0.1788,0.11996,-0.471287,0.0956538,0.0974073,0.0916145,0.120789,0.273004,0.0871263,0.0820581,-0.156573,0.174548,-0.410881,-0.0695121,0.451325,0.157792,-0.363941,-0.0610669,0.285096,-0.0453709,-0.0433942,0.0430098,0.0382714,0.146085,0.0390816,0.0365254,-0.217529,-0.139598,-0.120643,-0.0903309,-0.0354376,-0.270037,0.172684,-0.11376,-0.198607,0.0323647,-0.078836,-0.119132,0.166394,-0.0281917,0.0262072,-0.0338941,-0.0391354,-0.0753773,0.166237,-0.106079,-0.127228,0.0855171,0.0548223,0.0986047,0.0137426,-0.0615741,-0.024106,-0.0642234,0.0591033,0.0145178,0.00865116,0.0116192,0.0126072,-0.00393406,0.00387763,-0.00294016,-0.0133888,-0.00177883,-0.00274909,0.00427858,0.0012178,-0.00308985],[0.0383439,0.0382306,0.0386236,0.0381451,0.0304114,0.164363,0.0369583,0.0385448,0.0380294,0.0507475,0.116511,0.0386905,0.0377571,0.0732505,0.0872544,0.21424,0.0373633,0.255748,0.103674,0.245607,0.0467072,0.230119,0.0379479,0.0379387,0.0376307,0.0353065,0.138459,0.0384971,0.0367666,0.219458,0.142993,0.124603,0.0951306,0.03746,0.285764,0.185889,0.126399,0.222408,0.0373557,0.0913678,0.141191,0.203082,0.0381701,0.0358304,0.047101,0.054881,0.10909,0.270175,0.182646,0.219072,0.158345,0.103689,0.192739,0.0437958,0.214124,0.0875665,0.238288,0.221769,0.0546114,0.0372754,0.0556041,0.0897973,0.0296117,0.0375836,0.0372534,0.190097,0.0279101,0.0493619,0.104364,0.0374413,0.126575],[8.67511e-07,1.50021e-06,1.73982e-06,2.76738e-06,7.99435e-05,0.00413909,0.00964912,0.0115003,0.0159943,0.0173034,0.0191215,0.0243299,0.0297567,0.0325573,0.0454511,0.0551293,0.0628236,0.0776103,0.128008,0.138392,0.191063,0.215381,0.231848,0.252708,0.253061,0.278375,0.29139,0.310018,0.320495,0.321583,0.328935,0.332932,0.342343,0.344142,0.344677,0.35291,0.368113,0.371865,0.386276,0.388224,0.3988,0.41259,0.460161,0.464521,0.471768,0.475786,0.489589,0.538362,0.561381,0.561404,0.58915,0.597001,0.608934,0.753682,0.773681,0.783095,0.787529,0.789848,0.790363,0.81647,0.834478,0.888347,0.894308,0.917825,0.937094,0.94385,0.949182,0.955587,0.967298,0.974053,0.980525],[17,17,16,17,15,17,17,17,17,16,17,17,16,17,16,17,17,17,17,17,17,17,17,17,17,17,17,17,16,17,17,17,17,17,17,17,17,17,17,17,16,17,17,14,15,17,17,17,17,17,16,17,17,17,17,17,16,16,17,17,16,17,16,17,17,17,17,16,17,17,17]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Exposure<\/th>\n      <th>Outcome<\/th>\n      <th>bxy<\/th>\n      <th>se<\/th>\n      <th>p<\/th>\n      <th>nsnp<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":15,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,15,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row"}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="conclusion" class="section level1">
<h1><span class="header-section-number">6</span> - Conclusion</h1>
<hr />
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<ul>
<li>GSMR analysis suggests that vitamin-D concentration has a causal effect on several traits like Hypertension or Alzheimer. This sould be verified using a bigger sample size and multiple testing correction.</li>
</ul>
</div>
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>

&nbsp;



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
