<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="by Yan Holtz - 09 February 2018" />


<title>GSMR</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vitamin-D GWAS in the UKB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_Pheno_Description.html">Pheno</a>
</li>
<li>
  <a href="2_GWAS.html">GWAS</a>
</li>
<li>
  <a href="6_LDSC.html">LDSC</a>
</li>
<li>
  <a href="3_SMR.html">SMR</a>
</li>
<li>
  <a href="4_GSMR.html">GSMR</a>
</li>
<li>
  <a href="5_XIA_et_al.html">Xia et al</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<a href="https://github.com/holtzy/" class="github-corner" aria-label="View source on Github"><svg width="80" height="80" viewBox="0 0 250 250" style="fill:#2ecc71; color:#fff; position: absolute; top: 0; border: 0; right: 0;" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">GSMR</h1>
<h4 class="author"><em>by <a href="https://github.com/holtzy/">Yan Holtz</a> - 09 February 2018</em></h4>

</div>


<br><br>
<blockquote>
<p>This document applies the <a href="%22http://cnsgenomics.com/software/gsmr/%22">gsmr tool</a> to our data: is blood pressure causal to one of the 15 diseases?</p>
</blockquote>
<div id="data" class="section level1">
<h1><span class="header-section-number">1</span> - Data</h1>
<hr />
<p>To run a GSMR analysis we need:</p>
<ul>
<li><p>A risk factor GWAS summary statistics. This file has been created for diastolic pressure in the GWAS part. I can read it here: <code>/home/y.holtz/BLOOD_GWAS/1_GWAS/GWAS_blood_pressure.ma</code></p></li>
<li><p>GWAS summary statistics for diseases we want to explain. Iâ€™m gonna consider several sources of information:</p>
<ul>
<li><p>23 diseases comming from a meta-analysis of the UKBiobank and the GERA consortium. Available online <a href="http://cnsgenomics.com/data.html">here</a>. Diseases available are: allergic rhinitis, asthma, cancer ,card depress, dermatophytosis, dia2, dyslipid, hemorrhoids, hernia_abdominopelvic, hyper, insomnia, iron deficiency, irritable bowel, macdegen, ostioa, ostiop, peptic_ulcers, psychiatric, pvd stress, sum diseases, varicose veins). File available here on Inode: <code>ibscratch/users/uqzzhu1/dataset/formatted_meta/combined_ukb_gera*</code></p></li>
<li><p>66 other diseases already used in the <a href="https://www.nature.com/articles/s41467-017-02317-2">GSMR paper by Zhihong et al</a>. These include Alzheimer, depression, neuroticsm, Schizophrenia, Type II Diabete and others. File available here: <code>ibscratch/users/uqzzhu1/dataset/formatted_meta/</code></p></li>
</ul></li>
</ul>
<pre class="bash"><code># First transfert localy
cd /Users/y.holtz/Desktop
scp uqyholtz@inode.qbi.uq.edu.au://ibscratch/users/uqzzhu1/dataset/formatted_meta/* .

# Then from locally to delta
cd /Users/y.holtz/Desktop
scp * y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/GWAS_SUMSTAT</code></pre>
<p>I need to reformat these files to respect the GSMR input format:</p>
<pre class="bash"><code>cd /shares/compbio/Group-Wray/YanHoltz/GWAS_SUMSTAT
# Put that in a script
for i in $(ls *gz) ; do
  echo $i
  name=$(echo $i | sed &#39;s/.gz//&#39;)
  zcat $i | cut -d&quot; &quot; -f2,4,5,7,8,9,10,11   &gt; $name
done
qsubshcom &quot;bash script_clean.sh&quot; 1 20G GSMRrun 10:00:00 &quot;&quot;
rm *gz</code></pre>
</div>
<div id="run-gsmr" class="section level1">
<h1><span class="header-section-number">2</span> - Run GSMR</h1>
<hr />
<pre class="bash"><code># Go to the good folder
cd /home/y.holtz/BLOOD_GWAS/3_GSMR

# Prepare a file that gives the location of every bfile (one per chromosome)
ls /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*_v2_HM3_QC.bed | sed &#39;s/.bed//&#39; &gt; gsmr_ref_data.txt

# prepare a file that gives the link to the GWAS result for the risk factor
echo &quot;blood_pressure /home/y.holtz/BLOOD_GWAS/1_GWAS/GWAS_blood_pressure.ma&quot; &gt; gsmr_exposure.txt

# prepare a file that gives the link to the GWAS result for the outcomes
for i in $(ls /shares/compbio/Group-Wray/YanHoltz/*) ; do a=$(echo $i | sed &#39;s/.*gera_//&#39; | sed &#39;s/_1000g.*//&#39;) ; echo $a $i ; done &gt; gsmr_outcome.txt

# run the program
tmp_command=&quot;gcta64 --mbfile gsmr_ref_data.txt --gsmr-file gsmr_exposure.txt gsmr_outcome.txt --gsmr-alg 0 --out gsmr_result_bloodPressure&quot;
qsubshcom &quot;$tmp_command&quot; 1 20G GSMRrun 10:00:00 &quot;&quot;
</code></pre>
</div>
<div id="results" class="section level1">
<h1><span class="header-section-number">3</span> - Results</h1>
<hr />
<p>First I transfert the result locally for further analysis.</p>
<pre class="bash"><code># Then from locally to delta
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/home/y.holtz/BLOOD_GWAS/3_GSMR/gsmr_result_bloodPressure.gsmr .</code></pre>
<p>Read it in R and show it:</p>
<pre class="r"><code>gsmr=read.table(&quot;0_DATA/gsmr_result_bloodPressure.gsmr&quot;, header = T)
gsmr</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Exposure"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["Outcome"],"name":[2],"type":["fctr"],"align":["left"]},{"label":["bxy"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["se"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["p"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["nsnp"],"name":[6],"type":["int"],"align":["right"]}],"data":[{"1":"blood_pressure","2":"ALLERGIC_RHINITIS","3":"-0.003058280","4":"0.00484326","5":"5.27746e-01","6":"172"},{"1":"blood_pressure","2":"ASTHMA","3":"0.000876266","4":"0.00444014","5":"8.43553e-01","6":"168"},{"1":"blood_pressure","2":"CANCER","3":"-0.005699730","4":"0.00431584","5":"1.86616e-01","6":"173"},{"1":"blood_pressure","2":"CARD","3":"0.031930200","4":"0.00433070","5":"1.66832e-13","6":"172"},{"1":"blood_pressure","2":"DEPRESS","3":"0.001008500","4":"0.00541236","5":"8.52184e-01","6":"172"},{"1":"blood_pressure","2":"DERMATOPHYTOSIS","3":"0.002472250","4":"0.00778041","5":"7.50672e-01","6":"172"},{"1":"blood_pressure","2":"DIA2","3":"0.014170300","4":"0.00586236","5":"1.56420e-02","6":"169"},{"1":"blood_pressure","2":"DYSLIPID","3":"0.032735200","4":"0.00396132","5":"1.41196e-16","6":"169"},{"1":"blood_pressure","2":"HEMORRHOIDS","3":"-0.000411223","4":"0.00592219","5":"9.44641e-01","6":"169"},{"1":"blood_pressure","2":"HERNIA_ABDOMINOPELVIC","3":"-0.006545980","4":"0.00576638","5":"2.56293e-01","6":"172"},{"1":"blood_pressure","2":"HYPER","3":"0.125359000","4":"0.00396717","5":"3.79126e-219","6":"157"},{"1":"blood_pressure","2":"INSOMNIA","3":"-0.003881080","4":"0.00916641","5":"6.72002e-01","6":"173"},{"1":"blood_pressure","2":"IRON_DEFICIENCY","3":"-0.006871980","4":"0.01029950","5":"5.04635e-01","6":"171"},{"1":"blood_pressure","2":"IRRITABLE_BOWEL","3":"0.000391383","4":"0.00808441","5":"9.61388e-01","6":"173"},{"1":"blood_pressure","2":"MACDEGEN","3":"0.003334760","4":"0.01063640","5":"7.53883e-01","6":"170"},{"1":"blood_pressure","2":"OSTIOA","3":"-0.002251000","4":"0.00435096","5":"6.04907e-01","6":"171"},{"1":"blood_pressure","2":"OSTIOP","3":"-0.007004720","4":"0.00795002","5":"3.78266e-01","6":"172"},{"1":"blood_pressure","2":"PEPTIC_ULCERS","3":"-0.003034260","4":"0.01117800","5":"7.86046e-01","6":"172"},{"1":"blood_pressure","2":"PSYCHIATRIC","3":"0.016241600","4":"0.00634906","5":"1.05242e-02","6":"173"},{"1":"blood_pressure","2":"PVD","3":"0.019980600","4":"0.00842606","5":"1.77264e-02","6":"172"},{"1":"blood_pressure","2":"STRESS","3":"0.006839000","4":"0.00957953","5":"4.75278e-01","6":"172"},{"1":"blood_pressure","2":"SUM_DISEASES","3":"0.034622700","4":"0.00233816","5":"1.30655e-49","6":"168"},{"1":"blood_pressure","2":"VARICOSE_VEINS","3":"0.007822180","4":"0.00890293","5":"3.79614e-01","6":"171"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>A barplot of the pvalues.</p>
<pre class="r"><code>library(tidyverse)
gsmr %&gt;% 
  arrange(desc(p)) %&gt;%
  mutate(Outcome=factor(Outcome, Outcome)) %&gt;%
  ggplot( aes(x=Outcome, y=-log10(p))) +
    geom_segment( aes(x=Outcome, xend=Outcome, y=-log10(p), yend=0), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(size=4, color=&quot;orange&quot;) +
    coord_flip() +
    theme_light() +
    theme( panel.grid.major.y = element_blank()) +
    ylab(&quot;-log10( pvalue )&quot;) +
    xlab(&quot;&quot;)</code></pre>
<p><img src="4_GSMR_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Have a look to the bxy: the effect of the risk factor on the disease:</p>
<pre class="r"><code>gsmr %&gt;% 
  arrange(desc(p)) %&gt;%
  mutate(Outcome=factor(Outcome, Outcome)) %&gt;%
  mutate(significance=ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;)) %&gt;%
  ggplot( aes(x=Outcome, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=Outcome, xend=Outcome, y=bxy-se, yend=bxy+se), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;)) +
    coord_flip() +
    theme_light() +
    theme( panel.grid.major.y = element_blank()) +
    ylab(&quot;-log10( pvalue )&quot;) +
    xlab(&quot;&quot;)</code></pre>
<p><img src="4_GSMR_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Et yo</p>
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>

&nbsp;



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
