<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Computin polygenic risk scores to predict vitamin-D concentration in another dataset</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vitamin-D GWAS in the UKB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_Pheno_Description.html">Pheno</a>
</li>
<li>
  <a href="2_GWAS.html">GWAS</a>
</li>
<li>
  <a href="6_LDSC.html">LDSC</a>
</li>
<li>
  <a href="3_SMR.html">SMR</a>
</li>
<li>
  <a href="7_PRS.html">PRS</a>
</li>
<li>
  <a href="9_Methylation.html">Methylation</a>
</li>
<li>
  <a href="10_Integration.html">Integration</a>
</li>
<li>
  <a href="5_XIA_et_al.html">Xia et al</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Computin polygenic risk scores to predict vitamin-D concentration in another dataset</h1>

</div>


<hr>
<p><em>Yan Holtz, Zhihong Zhu, Julanne Frater, Perry Bartlett, Jian Yang, John McGrath</em></p>
<hr>
<p><br><br></p>
<p>This document describes how to compute Polygenic Risk Score (PRS) of the UKB sample. It also describes this PRS, and notably observe its geographical distribution.</p>
<div id="data" class="section level1">
<h1>Data</h1>
<hr />
<p>The idea is to use the effect of each SNP on the Vitamin-D as provided by <a href="https://www.nature.com/articles/s41467-017-02662-2">Xia et al.</a>. In order to do so we need:</p>
<ul>
<li>The GWAS summary statistics of <a href="https://www.nature.com/articles/s41467-017-02662-2">Xia et al.</a>. Available under different format in Delta: <code>/shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD</code><br />
</li>
<li>The genotypes of UKB participants: in delta <code>/afm01/Q0286/UKBiobank/v2EURu_HM3</code></li>
</ul>
<pre class="r"><code>library(tidyverse)</code></pre>
</div>
<div id="compute-prs" class="section level1">
<h1>Compute PRS</h1>
<hr />
<p>PRS are computed using the <code>--score</code> option of <a href="https://www.cog-genomics.org/plink2">plink version 1.9</a>. This work is made twice: first without any correction in the model, second correcting with PCs.</p>
<pre class="bash"><code># Good folder on delta
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS

# Need upper case for Xia et al sumstat, and keep only 3 columns: SNP id, allele, beta.
more /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.linear | awk &#39;{ print $2,toupper($4),$7 }&#39; | sed &#39;1d&#39; &gt; tmp

# prepare command
tmp_command=&quot;~/bin/plink --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr{TASK_ID}_v2_HM3_QC --score tmp --out output_PRS{TASK_ID}&quot;

# Send command in qsub
qsubshcom &quot;$tmp_command&quot; 1 30G PRS_plink 10:00:00 &quot;-array=1-22&quot;

# Now I&#39;ve got one PRS per chromosome.
# To have my final PRS, I have to multiply these PRS by the number of SNP per chromosome and sum them.

# Compute the number of SNP per chromosome:
for i in /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*_v2_HM3_QC*bim; do mylen=$(wc -l $i) ; mychr=$(echo $i | sed &#39;s/.*chr//&#39; | sed &#39;s/_.*//&#39;) ; echo $mychr $mylen ; done &gt; nbr_snp_per_chromosome

# In R I prepare my final file
R
size &lt;- read.table(&quot;nbr_snp_per_chromosome&quot;, sep=&quot; &quot;)
final &lt;- 0
for(i in seq(1,22)){
  print(i)
  chr &lt;- 1
  data &lt;- read.table(paste(&quot;output_PRS&quot;, chr, &quot;.profile&quot;, sep=&quot;&quot;), header=T)
  data$SCORE &lt;- data$SCORE * size[ which(size$V1 == chr) , 2]
  final &lt;- final + data$SCORE
}
output=data.frame(ID=data$FID, PRS=final)
write.table(output, file=&quot;VitaminD_PRS.txt&quot;, row.names=FALSE, quote=FALSE)</code></pre>
</div>
<div id="add-gps-coordinate" class="section level1">
<h1>- Add GPS Coordinate</h1>
<hr />
<p>I add the GPS coordinate of people and a few other information about them.</p>
<p>On Inode, I read the phenotype file and export the GPS positions:</p>
<pre class="r"><code># good repertory
ssh uqyholtz@inode.qbi.uq.edu.au
cd /ibscratch/wrayvisscher/Yan_Holtz/0_DATA 

# Get the good column + repeat the ID + change header to respect plink format
zcat /references/UKBiobank/pheno/download/9280_12505_UKBiobank.tab.gz | awk &#39;{print $1, $8, $144, $145, $146, $147, $148, $149}&#39; &gt; GPS_location.txt
gzip GPS_location.txt

# Transfer on Delta
cd ~/Desktop
scp uqyholtz@inode.qbi.uq.edu.au:/ibscratch/wrayvisscher/Yan_Holtz/0_DATA/GPS_location.txt.gz .
scp GPS_location.txt.gz y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/DATA/PHENOTYPE/UKB</code></pre>
<p>Merge both information</p>
<pre class="r"><code>cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS
R
library(dplyr)
PRS &lt;- read.table(&quot;VitaminD_PRS.txt&quot;, header=T)
GPS &lt;- read.table(&quot;/shares/compbio/Group-Wray/YanHoltz/DATA/PHENOTYPE/UKB/GPS_location.txt.gz&quot;, header=T)

# GPS location are distributed in 3 columns. In have to summarize this info in 1 column only.
GPS$latitude &lt;- GPS$f.129.0.0
GPS$latitude[is.na(GPS$latitude)] &lt;- GPS$f.129.1.0[is.na(GPS$latitude)]
GPS$latitude[is.na(GPS$latitude)] &lt;- GPS$f.129.2.0[is.na(GPS$latitude)]
GPS$longitude &lt;- GPS$f.130.0.0
GPS$longitude[is.na(GPS$longitude)] &lt;- GPS$f.130.1.0[is.na(GPS$longitude)]
GPS$longitude[is.na(GPS$longitude)] &lt;- GPS$f.130.2.0[is.na(GPS$longitude)]
GPS &lt;- GPS %&gt;% dplyr::select(f.eid, f.31.0.0, latitude, longitude)

# Merge
data &lt;- merge(PRS, GPS, by.x=&quot;ID&quot;, by.y=&quot;f.eid&quot;, all.x=T) %&gt;% na.omit() %&gt;% dplyr::select(longitude, latitude, PRS, ID, f.31.0.0)
write.table(data, file=&quot;VitaminD_PRS_with_GPS.txt&quot;, row.names=FALSE, quote=FALSE)
gzip VitaminD_PRS_with_GPS.txt
 
# Locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/VitaminD_PRS_with_GPS.txt.gz .</code></pre>
</div>
<div id="description-of-prs" class="section level1">
<h1>Description of PRS</h1>
<hr />
<p>Distribution of the PRS:</p>
<pre class="r"><code># Load the file
data &lt;- read.table(&quot;0_DATA/VitaminD_PRS_with_GPS.txt.gz&quot;, header=T)

# Distribution?
data %&gt;%
  mutate(sex = recode( f.31.0.0, `1`=&quot;female&quot;, `0`=&quot;male&quot;) ) %&gt;%
  ggplot(aes(x=PRS, fill=sex)) +
      geom_histogram( bins=100) +
      facet_wrap( ~sex, ncol=1 )</code></pre>
<p><img src="7_PRS_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="relationship-between-prs-and-diseases" class="section level1 tabset">
<h1>Relationship between PRS and diseases</h1>
<hr />
<p>I build a file with the ICD10 code obsered in UKB people. It uses the field <code>main ICD10</code> (field 41202-0.0 to 41202-0.379) and <code>secondary ICD10</code> (field 41204-0.0 to 41204-0.434).</p>
<pre class="r"><code># Work done in October 2017 on Inode

# Make script extract.sh:
zcat /references/UKBiobank/pheno/download/9280_12505_UKBiobank.tab.gz | cut -f1,8,9,22-24,144-146,147-149,4069-4071,4081,4082,4066-4068,3438,3439-3441,5899-5901,5902-5943,6181,6185-6564,6593-7027,7058-7062,26-28,29-31,4075-4077,4081,5896-5898,6008-6010,6179,6180,262-264,361-363,6593-7027 &gt; /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/DATA/UKB_phenos.txt

# Send with Qsub
qsub -b y -cwd -e tmp -o tmp &quot;/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/SCRIPTS/extract.sh&quot;

# Zip the result
gzip /ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/DATA/UKB_phenos.txt

# Transfer it locally
cd /Users/y.holtz/Desktop
scp uqyholtz@inode.qbi.uq.edu.au:/ibscratch/wrayvisscher/Yan_Holtz/3_UKB_COMO/DATA/UKB_phenos.txt.gz .

# Transfert on Delta
scp UKB_phenos.txt.gz y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/DATA/PHENOTYPE/UKB</code></pre>
<p>In this file, I can easily spot people who developped a disease using the ICD10 code of this disease. For example, let’s get the ids of people who developped Myeloid, ICD10 code C92*. Only 570 people are concerned..</p>
<pre class="r"><code>cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/LINK_DISEASE
printf &quot;F00\nG30\nG47\nD50\nF840&quot; &gt; tocheck
for i in $(cat tocheck) ; do 
  echo $i ; 
  zcat /shares/compbio/Group-Wray/YanHoltz/DATA/PHENOTYPE/UKB/UKB_phenos.txt.gz | grep $i | cut -f1 &gt; list_$i
done</code></pre>
<p>And a list of all ids in the UKB:</p>
<pre class="r"><code>zcat /shares/compbio/Group-Wray/YanHoltz/DATA/PHENOTYPE/UKB/UKB_phenos.txt.gz | cut -f1 &gt; list_all</code></pre>
<p>Make a summary file:</p>
<pre class="r"><code># Read all ids
data &lt;- read.table(&quot;list_all&quot;, header=TRUE)

# loop
for( i in c(&quot;F00&quot;, &quot;G30&quot;, &quot;G47&quot;, &quot;D50&quot;, &quot;F840&quot; ) ){
  print(i)
  tmp &lt;- read.table(paste(&quot;list_&quot;,i,sep=&quot;&quot;), header=FALSE)
  data$new &lt;- ifelse(data$f.eid %in% tmp$V1, 1, 0)
  colnames(data)[ which(colnames(data)==&quot;new&quot;)] = i
}</code></pre>
<p>Merge this info with the PRS data</p>
<pre class="r"><code>PRS &lt;- read.table(&quot;/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/PRS_general/VitaminD_PRS_with_GPS.txt.gz&quot;, header=TRUE)
data &lt;- merge(PRS, data, by.x=&quot;ID&quot;, by.y=&quot;f.eid&quot;, all.x=TRUE)

# Save file
write.table(data, file=&quot;PRS_with_diseases.txt&quot;, quote=FALSE, row.names=FALSE)

# zip it
gzip PRS_with_diseases.txt</code></pre>
<p>Send this file locally</p>
<pre class="r"><code># Transfer it locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/LINK_DISEASE/PRS_with_diseases.txt.gz .</code></pre>
<p>Visualize that:</p>
<pre class="r"><code>data &lt;- read.table(&quot;0_DATA/PRS_with_diseases.txt.gz&quot;, header=T)

data %&gt;% 
  dplyr::select(PRS, F00:F840) %&gt;%
  gather(disease, value, -1) %&gt;%
  ggplot( aes(x=as.factor(value), y=PRS, fill=as.factor(value))) +
    geom_boxplot() +
    theme(legend.position=&quot;none&quot;) +
    facet_wrap(~as.factor(disease))</code></pre>
<p><img src="7_PRS_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>Representation using decile: I count the number of people with a disease in each decile of Vitamin-D PRS. Looks like it work for F00 = Dementia in Alzheimer’s disease. But it is in the wrong side? If I have more vitamin D (decile is high), then I have more people witht this disease? This is very weird..</p>
<pre class="r"><code>data %&gt;%
    mutate(decile = ntile(PRS, 10)) %&gt;%
    dplyr::select(decile, F00:F840) %&gt;%
    gather(disease, value, -1) %&gt;%   
    group_by(decile, disease) %&gt;%
    summarize(nb=sum(value)) %&gt;%
    ggplot( aes(x=decile, y=nb)) +
      geom_bar(stat=&quot;identity&quot;, width=0.4, fill=&quot;skyblue&quot;, alpha=0.9) +
      scale_x_continuous(breaks=seq(1,10)) +
      xlab(&quot;VitaminD PRS decile&quot;) +
      ylab(&quot;Number of people with Myeloid&quot;) +
      facet_wrap(~disease, scale=&quot;free_y&quot;)</code></pre>
<p><img src="7_PRS_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="relationship-between-prs-and-pcs" class="section level1 tabset">
<h1>Relationship between PRS and PCs</h1>
<hr />
<p>First of all I need to get the coordinate of each UKB participants on its 10 first PCs. I’m gonna get the PCs in Angli file:</p>
<pre class="r"><code>cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/LINK_PC
cp /gpfs/gpfs01/polaris/Q0286/uqaxue/phen/covar_sex_age_10PCs.txt .</code></pre>
<p>Then I merge my PRS with this PC information</p>
<pre class="r"><code>PRS &lt;- read.table(&quot;/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/PRS_general/VitaminD_PRS_with_GPS.txt.gz&quot;, header=TRUE)
PC &lt;- read.table(&quot;covar_sex_age_10PCs.txt&quot;, header=T)

# Merge
data &lt;- merge(PRS, PC, by.x=&quot;ID&quot;, by.y=&quot;FID&quot;, all.x=TRUE)

# Save file
write.table(data, file=&quot;PRS_with_PC.txt&quot;, quote=FALSE, row.names=FALSE)

# zip it
gzip PRS_with_PC.txt</code></pre>
<p>Send this file locally</p>
<pre class="r"><code># Transfer it locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/LINK_PC/PRS_with_PC.txt.gz .</code></pre>
<p>Check the relationship</p>
<pre class="r"><code>data &lt;- read.table(&quot;0_DATA/PRS_with_PC.txt.gz&quot;, header=TRUE)

data %&gt;% head(1000) %&gt;%
  dplyr::select(PRS, PC1:PC10) %&gt;%
  gather(PC, value, -1) %&gt;%
  ggplot( aes(x=PRS, y=value)) +
    stat_density_2d(aes(fill = ..density..), geom = &quot;raster&quot;, contour = FALSE) +
    scale_fill_distiller(palette= &quot;Spectral&quot;, direction=1) +
    scale_x_continuous(expand = c(0, 0)) +
    scale_y_continuous(expand = c(0, 0)) +
    theme(
      legend.position=&#39;none&#39;
    ) +
    facet_wrap( ~PC) +
    ggtitle(&quot;Correlation between VitaminD PRS and the 10 first PCs&quot;) +
    theme_classic()</code></pre>
<p><img src="7_PRS_files/figure-html/unnamed-chunk-17-1.png" width="960" /></p>
<p>Check through a Scatterplot</p>
<pre class="r"><code>data %&gt;% 
  dplyr::select(PRS, PC1:PC10) %&gt;%
  gather(PC, value, -1) %&gt;%
  ggplot( aes(x=PRS, y=value)) +
    geom_point(size=0.1, alpha=0.1) +
    theme(
      legend.position=&#39;none&#39;
    ) +
    facet_wrap( ~PC) +
    ggtitle(&quot;Correlation between VitaminD PRS and the 10 first PCs&quot;) </code></pre>
<p><img src="7_PRS_files/figure-html/unnamed-chunk-18-1.png" width="960" /></p>
<p>Clearly something happening on PCs 1, 2 3, 4, 6</p>
</div>
<div id="geographical-distribution-of-prs" class="section level1">
<h1>Geographical Distribution of PRS</h1>
<hr />
<p>I used the <a href="https://holtzyan.shinyapps.io/UKB_geo/">UKB geo application</a> to represent the geographical distribution of the PRS. I used the place of birth has a proxy of the position of each UKB study participant.</p>
<p>There is no organization of the PRS per region.</p>
<p><img src="IMG/PRS_map1.png" alt="map1" /> <img src="IMG/PRS_map2.png" alt="map2" /></p>
</div>
<div id="geographical-distribution-of-the-gc-gene" class="section level1">
<h1>Geographical distribution of the GC gene</h1>
<hr />
<p>First I need to get the allele of the genome wide significant SNPs for all the UK Biobank participants. I’ve got 6 SNPs to recover as described by Xia et al.</p>
<pre class="r"><code>cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/PRS_GCgene
tmp_command=&quot;plink --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr4_v2_HM3_QC --snp rs3755967 --recode --out allele_at_GC&quot; ; qsubshcom &quot;$tmp_command&quot; 1 10G recover_allele 10:00:00 &quot;&quot;
tmp_command=&quot;plink --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr11_v2_HM3_QC --snp rs12785878 --recode --out allele_at_DHCR7&quot; ; qsubshcom &quot;$tmp_command&quot; 1 10G recover_allele 10:00:00 &quot;&quot;

# Other SNP are not in HAPMAP3
tmp_command=&quot;plink --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_impQC/ukbEURu_imp_chr11_v2_imp_QC --snp rs10741657 --recode --out allele_at_CYP2R1&quot; ; qsubshcom &quot;$tmp_command&quot; 1 10G recover_allele 10:00:00 &quot;&quot;
tmp_command=&quot;plink --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_impQC/ukbEURu_imp_chr20_v2_imp_QC --snp rs17216707 --recode --out allele_at_CYP24A1&quot; ; qsubshcom &quot;$tmp_command&quot; 1 10G recover_allele 10:00:00 &quot;&quot;
tmp_command=&quot;plink --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_impQC/ukbEURu_imp_chr12_v2_imp_QC --snp rs10745742 --recode --out allele_at_AMDHD1&quot; ; qsubshcom &quot;$tmp_command&quot; 1 10G recover_allele 10:00:00 &quot;&quot;
tmp_command=&quot;plink --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_impQC/ukbEURu_imp_chr14_v2_imp_QCC --snp rs8018720 --recode --out allele_at_SEC23A&quot; ; qsubshcom &quot;$tmp_command&quot; 1 10G recover_allele 10:00:00 &quot;&quot;</code></pre>
<p>Then I need to merge this file with the localization of people.</p>
<pre class="r"><code>R
library(dplyr)
PRS &lt;- read.table(&quot;allele_at_GC.ped&quot;, header=F)
GPS &lt;- read.table(&quot;/shares/compbio/Group-Wray/YanHoltz/DATA/PHENOTYPE/UKB/GPS_location.txt.gz&quot;, header=T)

# GPS location are distributed in 3 columns. In have to summarize this info in 1 column only.
GPS$latitude &lt;- GPS$f.129.0.0
GPS$latitude[is.na(GPS$latitude)] &lt;- GPS$f.129.1.0[is.na(GPS$latitude)]
GPS$latitude[is.na(GPS$latitude)] &lt;- GPS$f.129.2.0[is.na(GPS$latitude)]
GPS$longitude &lt;- GPS$f.130.0.0
GPS$longitude[is.na(GPS$longitude)] &lt;- GPS$f.130.1.0[is.na(GPS$longitude)]
GPS$longitude[is.na(GPS$longitude)] &lt;- GPS$f.130.2.0[is.na(GPS$longitude)]
GPS &lt;- GPS %&gt;% dplyr::select(f.eid, f.31.0.0, latitude, longitude)

# Merge &amp; clean
data &lt;- merge(PRS, GPS, by.x=&quot;V1&quot;, by.y=&quot;f.eid&quot;, all.x=T) %&gt;% 
  na.omit() %&gt;%
  mutate(GC_gene=as.factor(paste(V7,V8,sep=&quot;&quot;))) %&gt;% 
  filter(GC_gene!=&quot;00&quot;) %&gt;% 
  droplevels() %&gt;% 
  mutate(GC=as.numeric(GC_gene)) %&gt;% 
  dplyr::select(latitude, longitude, GC)
  
# Save and zip
write.table(data, file=&quot;VitaminD_PRS_geneGC.txt&quot;, row.names=FALSE, quote=FALSE)
gzip VitaminD_PRS_geneGC.txt
 
# Locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/7_PRS/PRS_GCgene/VitaminD_PRS_geneGC.txt.gz .</code></pre>
Result:
<center>
<img src="IMG/PRS_GC_gene.png" />
</center>
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>

&nbsp;



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
