<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="by Yan Holtz - 03 May 2018" />


<title>mtCOJO</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vitamin-D GWAS in the UKB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_Pheno_Description.html">Pheno</a>
</li>
<li>
  <a href="2_GWAS.html">GWAS</a>
</li>
<li>
  <a href="6_LDSC.html">LDSC</a>
</li>
<li>
  <a href="3_SMR.html">SMR</a>
</li>
<li>
  <a href="4_GSMR.html">GSMR</a>
</li>
<li>
  <a href="7_PRS.html">PRS</a>
</li>
<li>
  <a href="8_mtCOJO.html">mtCOJO</a>
</li>
<li>
  <a href="9_Methylation.html">Methylation</a>
</li>
<li>
  <a href="5_XIA_et_al.html">Xia et al</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<a href="https://github.com/holtzy/" class="github-corner" aria-label="View source on Github"><svg width="80" height="80" viewBox="0 0 250 250" style="fill:#2ecc71; color:#fff; position: absolute; top: 0; border: 0; right: 0;" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">mtCOJO</h1>
<h4 class="author"><em>by <a href="https://github.com/holtzy/">Yan Holtz</a> - 03 May 2018</em></h4>

</div>


<p><br><br></p>
<blockquote>
<p>This document applies the <a href="%22http://cnsgenomics.com/software/gcta/#mtCOJO%22">mtCOJO tool</a> to our data. It allows to check if the causal effect of Vitamin-D on several traits can be biaised by other risk factors</p>
</blockquote>
<div id="method" class="section level1">
<h1><span class="header-section-number">1</span> - Method</h1>
<hr />
<p>mtCOJO allows to build a new GWAS summary statistics for a trait using summary statistics of one or several other risk factors. Here are the risk factor we considere in our study:</p>
<ul>
<li>Burning on UV exposure) (Bonilla et al. 2013)</li>
<li>Dark skin (Bonilla et al. 2013)</li>
<li>Outdoor activity (physical activity)</li>
<li>BMI</li>
<li>Smoking</li>
</ul>
<p>We considered these risk factors one by one in a first step, and then considered them all together.</p>
<p>The input of mtCOJO are:</p>
<ul>
<li><strong>mbfile</strong>: individual level genotypes of a reference sample. We will use the Hapmap3 SNPs of the UKB population. This is used for LD estimation.</li>
<li><strong>mtcojo-file</strong>: a list of the GWAS summary data and prevalence of diseases. first row is the target trait. Other rows are covariate.</li>
<li><strong>LD score references</strong>: come from the <a href="https://github.com/bulik/ldsc">LDSC tool</a>.</li>
</ul>
</div>
<div id="run-mtcojo" class="section level1">
<h1><span class="header-section-number">2</span> - Run mtCOJO</h1>
<hr />
<p>This script run mtCOJO. The target trait is VitaminD. The related risk factor is height. It gives me as output a new GWAS summary statistics corrected for height.</p>
<pre class="r"><code># Specific repo
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO

# Build mbfile
ls  /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*bed | sed &#39;s/.bed//&#39; &gt; mtcojo_mbfile.txt

# Build mtcojo file. One per risk factor

# BMI
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_BMI.txt
echo &quot;BMI /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/height_giant_2014.txt&quot; &gt;&gt; list_of_trait_BMI.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_BMI.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_BMI_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# SMOKE
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_SMOKE.txt
echo &quot;SMOKE /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/ukbEUR_SI_cojo.txt&quot; &gt;&gt; list_of_trait_SMOKE.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_SMOKE.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_SMOKE_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# ALL
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_ALL.txt
echo &quot;BMI /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/height_giant_2014.txt&quot; &gt;&gt; list_of_trait_ALL.txt
echo &quot;SMOKE /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/ukbEUR_SI_cojo.txt&quot; &gt;&gt; list_of_trait_ALL.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_ALL.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_ALL_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;</code></pre>
<p>I now have a several files called <code>mtcojo_*_result_vitaminD_result.mtcojo.cma</code> that are a GWAS summary statistics for Vitamin-D corrected for all my Risk factors.</p>
</div>
<div id="run-gsmr" class="section level1">
<h1><span class="header-section-number">3</span> - Run GSMR</h1>
<hr />
<p>I can re run GSMR on this new GWAS summary statistics:</p>
<pre class="r"><code>myRiskFact=$(echo &quot;BMI&quot;)

# Specific repo
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/
mkdir GSMR_$myRiskFact
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/GSMR_$myRiskFact

# Prepare a file that gives the location of every b-file (one per chromosome)
ls /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*_v2_HM3_QC.bed | sed &#39;s/.bed//&#39; &gt; gsmr_ref_data.txt

# prepare a file that gives the link to the GWAS result for the risk factor = Vitamin D. After correction for risk factors
cat /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/mtcojo_${myRiskFact}_result_vitaminD.mtcojo.cma | cut -f1-8 &gt; file.ma
a=$(pwd)
echo &quot;vitaminD $a/file.ma&quot; &gt; gsmr_exposure.txt

# prepare ONE file that lists all the outcomes. This file has been build in Excel format locally
cp /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR/gsmr_outcome.txt .

# Split this file: one file per outcome:
split -l 1 --numeric-suffixes gsmr_outcome.txt 
for i in x0* ; do a=$(echo $i | sed &#39;s/x0/x/&#39;) ; mv $i $a ; done

# send an array of GSMR
tmp_command=&quot;gcta64 --mbfile gsmr_ref_data.txt --gsmr-file gsmr_exposure.txt x{TASK_ID}  --gsmr-direction 0 --out gsmr_aftermtcojo${myRiskFact}_result_vitaminDXiaEtAl_{TASK_ID}&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO_GSMR_array 10:00:00 &quot;-array=1-72&quot;



# Once it&#39;s over, concatenate the results in a unique file
myRiskFact=$(echo &quot;SMOKE&quot;)
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/GSMR_${myRiskFact}
cat gsmr_aftermtcojo*gsmr | head -1 &gt; gsmr_aftermtcojo${myRiskFact}_result_vitaminDXiaEtAl.gsmr
cat gsmr_aftermtcojo*gsmr | grep -v &quot;Exposure&quot; &gt;&gt;  gsmr_aftermtcojo${myRiskFact}_result_vitaminDXiaEtAl.gsmr
rm *[0-9].gsmr x* *badsnps qsub* *log</code></pre>
<p>Transfer the results locally for further analysis.</p>
<pre class="bash"><code># Then from locally to delta
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/GSMR*/gsmr_aftermtcojo*_result_vitaminDXiaEtAl.gsmr .</code></pre>
</div>
<div id="result" class="section level1 tabset">
<h1><span class="header-section-number">4</span> - Result</h1>
<hr />
<pre class="r"><code># A function that format a gsmr result
plot_primary &lt;- function(file){

  # Read gsmr result after mtcojo
  gsmr=read.table(file, header = T)
  
  # Show the information of GSMR withour mtCOJO
  before &lt;- read.table(&quot;0_DATA/gsmr_result_vitaminDXiaEtAl.gsmr&quot;, header = T)
  before &lt;- before %&gt;%
      dplyr::select(Outcome, bxy, p)
  colnames(before) &lt;- c(&quot;Outcome&quot;, &quot;bxy_before&quot;, &quot;p_before&quot;)
  
  # Merge
  gsmr &lt;- merge(gsmr, before, by.x=&quot;Outcome&quot;, by.y=&quot;Outcome&quot;, all.x=T)
  
  # Read the meaning of files:
  meaning=read.table(&quot;0_DATA/list_of_traits_GSMR.csv&quot;, header=T, sep=&quot;,&quot;)
  primary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Main==&quot;X&quot;) ] )
  secondary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Main!=&quot;X&quot;) ] )
  
  # Compute thresholds
  thres_primary &lt;- 0.05 / nrow(primary)
  thres_secondary &lt;- 0.05 / nrow(secondary)

  primary %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_primary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  mutate(significance_before=ifelse(p_before&lt;thres_primary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=name, xend=name, y=bxy-se, yend=bxy+se), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    geom_point(aes(y=bxy_before), size=2, shape=20) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_bw() +
    theme( panel.grid.major.y = element_line(size=0.1)) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;)
}


plot_secondary &lt;- function(file){

  # Read gsmr result after mtcojo
  gsmr=read.table(file, header = T)
  
  # Show the information of GSMR withour mtCOJO
  before &lt;- read.table(&quot;0_DATA/gsmr_result_vitaminDXiaEtAl.gsmr&quot;, header = T)
  before &lt;- before %&gt;%
      dplyr::select(Outcome, bxy, p)
  colnames(before) &lt;- c(&quot;Outcome&quot;, &quot;bxy_before&quot;, &quot;p_before&quot;)
  
  # Merge
  gsmr &lt;- merge(gsmr, before, by.x=&quot;Outcome&quot;, by.y=&quot;Outcome&quot;, all.x=T)
  
  # Read the meaning of files:
  meaning=read.table(&quot;0_DATA/list_of_traits_GSMR.csv&quot;, header=T, sep=&quot;,&quot;)
  primary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Main==&quot;X&quot;) ] )
  secondary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Main!=&quot;X&quot;) ] )
  
  # Compute thresholds
  thres_primary &lt;- 0.05 / nrow(primary)
  thres_secondary &lt;- 0.05 / nrow(secondary)

  secondary %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_secondary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  mutate(significance_before=ifelse(p_before&lt;thres_secondary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=name, xend=name, y=bxy-se, yend=bxy+se), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    geom_point(aes(y=bxy_before), size=2, shape=20) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_bw() +
    theme( panel.grid.major.y = element_line(size=0.1)) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;)
}</code></pre>
<div id="bmi" class="section level2">
<h2><span class="header-section-number">4.1</span> BMI</h2>
<p><strong>Primary traits</strong></p>
<pre class="r"><code>plot_primary(&quot;0_DATA/gsmr_aftermtcojoBMI_result_vitaminDXiaEtAl.gsmr&quot;)</code></pre>
<p><img src="8_mtCOJO_files/figure-html/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
<p><strong>Secondary traits</strong></p>
<pre class="r"><code>plot_secondary(&quot;0_DATA/gsmr_aftermtcojoBMI_result_vitaminDXiaEtAl.gsmr&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="8_mtCOJO_files/figure-html/unnamed-chunk-7-1.png" alt="Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size. Black points shows the effect size before correction using mtCOJO" width="768" />
<p class="caption">
Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size. Black points shows the effect size before correction using mtCOJO
</p>
</div>
</div>
<div id="smoke" class="section level2">
<h2><span class="header-section-number">4.2</span> Smoke</h2>
<p><strong>Primary traits</strong></p>
<pre class="r"><code>plot_primary(&quot;0_DATA/gsmr_aftermtcojoSMOKE_result_vitaminDXiaEtAl.gsmr&quot;)</code></pre>
<p><img src="8_mtCOJO_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /></p>
<p><strong>Secondary traits</strong></p>
<pre class="r"><code>plot_secondary(&quot;0_DATA/gsmr_aftermtcojoSMOKE_result_vitaminDXiaEtAl.gsmr&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="8_mtCOJO_files/figure-html/unnamed-chunk-9-1.png" alt="Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size. Black points shows the effect size before correction using mtCOJO" width="768" />
<p class="caption">
Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size. Black points shows the effect size before correction using mtCOJO
</p>
</div>
</div>
<div id="dark-skin-color" class="section level2">
<h2><span class="header-section-number">4.3</span> Dark skin color</h2>
</div>
<div id="outdoor-activity" class="section level2">
<h2><span class="header-section-number">4.4</span> Outdoor activity</h2>
</div>
<div id="burning-on-uv-exposure" class="section level2">
<h2><span class="header-section-number">4.5</span> Burning on UV exposure</h2>
</div>
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>

&nbsp;



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
