<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="by Yan Holtz - 01 May 2018" />


<title>Using Methylation data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vitamin-D GWAS in the UKB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_Pheno_Description.html">Pheno</a>
</li>
<li>
  <a href="2_GWAS.html">GWAS</a>
</li>
<li>
  <a href="6_LDSC.html">LDSC</a>
</li>
<li>
  <a href="3_SMR.html">SMR</a>
</li>
<li>
  <a href="4_GSMR.html">GSMR</a>
</li>
<li>
  <a href="7_PRS.html">PRS</a>
</li>
<li>
  <a href="8_mtCOJO.html">mtCOJO</a>
</li>
<li>
  <a href="9_Methylation.html">Methylation</a>
</li>
<li>
  <a href="5_XIA_et_al.html">Xia et al</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<a href="https://github.com/holtzy/" class="github-corner" aria-label="View source on Github"><svg width="80" height="80" viewBox="0 0 250 250" style="fill:#2ecc71; color:#fff; position: absolute; top: 0; border: 0; right: 0;" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Using Methylation data</h1>
<h4 class="author"><em>by <a href="https://github.com/holtzy/">Yan Holtz</a> - 01 May 2018</em></h4>

</div>


<p><br><br></p>
<blockquote>
<p>This document applies the SMR tools to methylation data.</p>
</blockquote>
<div id="method" class="section level1">
<h1><span class="header-section-number">1</span> - Method</h1>
<hr />
<p>I’m gonna use the methylation data located on Delta at <code>/gpfs/gpfs01/polaris/Q0286/uqywu16/omics/methy</code></p>
<p>I run SMR on it, like I’ve done for eQTL data.</p>
</div>
<div id="dna-methylation-data-description" class="section level1 tabset">
<h1><span class="header-section-number">2</span> - DNA methylation data description</h1>
<hr />
<pre class="r"><code># load data
data &lt;- read.table(&quot;0_DATA/smr_methyl_VitaminDXiaEtAl.smr&quot;, header=T)</code></pre>
<p>The data set is composed by <strong>85334 SNP - gene methylation associations</strong>. These association are distributed along every chromosomes:</p>
<pre class="r"><code>table(data$ProbeChr) %&gt;%
  as.data.frame() %&gt;%
  ggplot( aes(x=Var1, y=Freq)) +
    geom_bar(stat=&quot;identity&quot;, fill=&quot;skyblue&quot;) +
    xlab(&quot;chromosome&quot;) +
    ylab(&quot;Number of SNP / gene methylation association&quot;) +
    theme_ipsum()</code></pre>
<p><img src="9_Methylation_files/figure-html/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="run-smr" class="section level1">
<h1><span class="header-section-number">3</span> - Run SMR</h1>
<hr />
<p>The first step of SMR consists to run the algorythm on every significant mQTL (mQTL = significant relationship between a SNP allele and the methylation of a gene). For each gene involved in an mQTL, we are going to test its putative effect on our trait (vitaminD) thanks to the GWAS summary statistic of this trait.</p>
<pre class="bash"><code># Good directory
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/9_METHYLATION

# Run the analysis
tmp_command=&quot;smr_Linux --bfile /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr{TASK_ID}_v2_HM3_QC --gwas-summary /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/1_GWAS/GWAS_vitaminD_XiaEtAL.ma --beqtl-summary /gpfs/gpfs01/polaris/Q0286/uqywu16/omics/methy/bl_meqtl_cis_std_chr{TASK_ID} --out smr_methyl_VitaminDXiaEtAl_{TASK_ID} --thread-num 1&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G smr_VitaminD 10:00:00 &quot;-array=1-22&quot;

# Concatenate chromosome results
cat smr_methyl_V*smr | head -1 &gt; smr_methyl_VitaminDXiaEtAl.smr
cat smr_methyl_V*smr | grep -v &quot;^probeID&quot; &gt;&gt; smr_methyl_VitaminDXiaEtAl.smr

# Transfer locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA
scp  y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/9_METHYLATION/smr_methyl_VitaminDXiaEtAl.smr  .</code></pre>
</div>
<div id="result" class="section level1 tabset">
<h1><span class="header-section-number">4</span> - Result</h1>
<hr />
<p>Significant SMR results are found in 3 genetic regions. These 3 regions are also detected by the Vitamin D GWAS.</p>
<pre class="r"><code># Compute signif threshold
thres &lt;- 0.05 / nrow(data)</code></pre>
<div id="chromo-4-gc" class="section level2">
<h2><span class="header-section-number">4.1</span> Chromo 4 (GC)</h2>
<pre class="r"><code># Show significant association:
data %&gt;% 
  filter(p_SMR &lt; thres &amp; topSNP_chr==4)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["probeID"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["ProbeChr"],"name":[2],"type":["int"],"align":["right"]},{"label":["Gene"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["Probe_bp"],"name":[4],"type":["int"],"align":["right"]},{"label":["topSNP"],"name":[5],"type":["fctr"],"align":["left"]},{"label":["topSNP_chr"],"name":[6],"type":["int"],"align":["right"]},{"label":["topSNP_bp"],"name":[7],"type":["int"],"align":["right"]},{"label":["A1"],"name":[8],"type":["fctr"],"align":["left"]},{"label":["A2"],"name":[9],"type":["fctr"],"align":["left"]},{"label":["Freq"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["b_GWAS"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["se_GWAS"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["p_GWAS"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["b_eQTL"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["se_eQTL"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["p_eQTL"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["b_SMR"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["se_SMR"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["p_SMR"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["p_HEIDI"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["nsnp_HEIDI"],"name":[21],"type":["int"],"align":["right"]}],"data":[{"1":"cg17429887","2":"4","3":"GC*","4":"72723478","5":"rs1491714","6":"4","7":"72704419","8":"C","9":"T","10":"0.913573","11":"0.0546","12":"0.0052","13":"1.904e-25","14":"0.468546","15":"0.0592018","16":"2.484665e-15","17":"0.116531","18":"0.0184381","19":"2.613709e-10","20":"2.236695e-07","21":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p><br></p>
<div class="figure">
<img src="IMG/VitaminD_XiaEtAl.rs3755967.png" />

</div>
<p>Can I make a diagram of these associations?</p>
<pre class="r"><code>tmp &lt;- data %&gt;%
  filter(ProbeChr==4) %&gt;%
  filter(Probe_bp&gt;71000000 &amp; Probe_bp&lt;74000000) %&gt;%
  select(Gene, topSNP, p_SMR, Probe_bp, topSNP_bp) %&gt;%
  gather(key, position, -1, -2, -3) %&gt;%
  mutate(position=position/1000000)
  
tmp %&gt;% ggplot( aes(y=position, x=key)) +
    geom_line(aes(group=paste(Gene,topSNP), color=ifelse(p_SMR&lt;thres,&quot;signif&quot;,&quot;non signif&quot;))) +
    geom_point() +
    geom_text( 
      data=subset(tmp, key==&quot;Probe_bp&quot;) %&gt;% select(Gene, key, position) %&gt;% group_by(Gene) %&gt;% summarize(key=unique(key), position=mean(position)), 
      aes( label=Gene) , angle=45, color=&quot;grey&quot;, hjust=1, nudge_x = -0.1
    ) +
    geom_text( 
      data=subset(tmp, key==&quot;topSNP_bp&quot;) %&gt;% select(topSNP, key, position) %&gt;% group_by(topSNP) %&gt;% summarize(key=unique(key), position=mean(position)), 
      aes( x=key, label=topSNP) , angle=45, color=&quot;grey&quot;, hjust=0, nudge_x = 0.1
    ) +
    coord_flip() +
    theme_ipsum() +
    xlab(&quot;&quot;) +
    ylab(&quot;Position (Mb)&quot;) +
    labs(color=&quot;SMR assoc with Vitamin-D&quot;)</code></pre>
<p><img src="9_Methylation_files/figure-html/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="chromo-11-14mb-cyp2r1" class="section level2">
<h2><span class="header-section-number">4.2</span> Chromo 11 ~ 14Mb (CYP2R1)</h2>
<pre class="r"><code># Show significant association:
data %&gt;% 
  filter(p_SMR &lt; thres &amp; topSNP_chr==11 &amp; topSNP_bp&lt;40000000) %&gt;%
  arrange(Probe_bp)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["probeID"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["ProbeChr"],"name":[2],"type":["int"],"align":["right"]},{"label":["Gene"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["Probe_bp"],"name":[4],"type":["int"],"align":["right"]},{"label":["topSNP"],"name":[5],"type":["fctr"],"align":["left"]},{"label":["topSNP_chr"],"name":[6],"type":["int"],"align":["right"]},{"label":["topSNP_bp"],"name":[7],"type":["int"],"align":["right"]},{"label":["A1"],"name":[8],"type":["fctr"],"align":["left"]},{"label":["A2"],"name":[9],"type":["fctr"],"align":["left"]},{"label":["Freq"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["b_GWAS"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["se_GWAS"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["p_GWAS"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["b_eQTL"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["se_eQTL"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["p_eQTL"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["b_SMR"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["se_SMR"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["p_SMR"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["p_HEIDI"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["nsnp_HEIDI"],"name":[21],"type":["int"],"align":["right"]}],"data":[{"1":"cg22106556","2":"11","3":"RRAS2","4":"14380825","5":"rs11023197","6":"11","7":"14380998","8":"G","9":"A","10":"0.647856","11":"0.0164","12":"0.0021","13":"1.416e-14","14":"0.734409","15":"0.0320909","16":"6.513048e-116","17":"0.0223309","18":"0.00302135","19":"1.456941e-13","20":"7.272002e-08","21":"20"},{"1":"cg19336497","2":"11","3":"RRAS2*","4":"14380999","5":"rs10832242","6":"11","7":"14340504","8":"T","9":"C","10":"0.645931","11":"0.0173","12":"0.0021","13":"4.922e-17","14":"1.226210","15":"0.0216580","16":"0.000000e+00","17":"0.0141086","18":"0.00173063","19":"3.572232e-16","20":"4.348532e-10","21":"20"},{"1":"cg01475538","2":"11","3":"RRAS2*","4":"14381234","5":"rs10832242","6":"11","7":"14340504","8":"T","9":"C","10":"0.645931","11":"0.0173","12":"0.0021","13":"4.922e-17","14":"-0.263433","15":"0.0347739","16":"3.574574e-14","17":"-0.0656712","18":"0.01177690","19":"2.457111e-08","20":"5.694381e-03","21":"20"},{"1":"cg04467639","2":"11","3":"COPB1","4":"14495049","5":"rs10832276","6":"11","7":"14483274","8":"C","9":"T","10":"0.631027","11":"0.0210","12":"0.0020","13":"4.416e-25","14":"1.264190","15":"0.0182259","16":"0.000000e+00","17":"0.0166115","18":"0.00160007","19":"3.003100e-25","20":"1.214631e-09","21":"20"},{"1":"cg25251204","2":"11","3":"CYP2R1*","4":"14927456","5":"rs1553006","6":"11","7":"14922950","8":"G","9":"A","10":"0.548406","11":"-0.0175","12":"0.0020","13":"6.077e-19","14":"0.633153","15":"0.0312926","16":"4.981466e-91","17":"-0.0276395","18":"0.00344152","19":"9.653631e-16","20":"2.421411e-10","21":"20"},{"1":"cg00085434","2":"11","3":"CYP2R1*","4":"14927489","5":"rs1553006","6":"11","7":"14922950","8":"G","9":"A","10":"0.548406","11":"-0.0175","12":"0.0020","13":"6.077e-19","14":"0.599996","15":"0.0310136","16":"2.192011e-83","17":"-0.0291668","18":"0.00365844","19":"1.555153e-15","20":"9.474278e-11","21":"20"},{"1":"cg16394018","2":"11","3":"COPB1","4":"14927549","5":"rs1553006","6":"11","7":"14922950","8":"G","9":"A","10":"0.548406","11":"-0.0175","12":"0.0020","13":"6.077e-19","14":"0.351599","15":"0.0320481","16":"5.270551e-28","17":"-0.0497727","18":"0.00727592","19":"7.878620e-12","20":"3.627764e-05","21":"20"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="figure">
<img src="IMG/VitaminD_XiaEtAl.rs10741657_large.png" />

</div>
<pre class="r"><code>tmp &lt;- data %&gt;%
  filter(ProbeChr==11) %&gt;%
  filter(Probe_bp&gt;14000000 &amp; Probe_bp&lt;15500000) %&gt;%
  select(Gene, topSNP, p_SMR, Probe_bp, topSNP_bp) %&gt;%
  gather(key, position, -1, -2, -3) %&gt;%
  mutate(position=position/1000000)
  
tmp %&gt;% ggplot( aes(y=position, x=key)) +
    geom_line(aes(group=paste(Gene,topSNP), color=ifelse(p_SMR&lt;thres,&quot;signif&quot;,&quot;non signif&quot;))) +
    geom_point() +
    geom_text( 
      data=subset(tmp, key==&quot;Probe_bp&quot;) %&gt;% select(Gene, key, position) %&gt;% group_by(Gene) %&gt;% summarize(key=unique(key), position=mean(position)), 
      aes( label=Gene) , angle=45, color=&quot;grey&quot;, hjust=1, nudge_x = -0.1
    ) +
    geom_text( 
      data=subset(tmp, key==&quot;topSNP_bp&quot;) %&gt;% select(topSNP, key, position) %&gt;% group_by(topSNP) %&gt;% summarize(key=unique(key), position=mean(position)), 
      aes( x=key, label=topSNP) , angle=45, color=&quot;grey&quot;, hjust=0, nudge_x = 0.1
    ) +
    coord_flip() +
    theme_ipsum() +
    xlab(&quot;&quot;) +
    ylab(&quot;Position (Mb)&quot;) +
    labs(color=&quot;SMR assoc with Vitamin-D&quot;)</code></pre>
<p><img src="9_Methylation_files/figure-html/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="chromo-11-71mb-dhcr7" class="section level2">
<h2><span class="header-section-number">4.3</span> Chromo 11 ~ 71Mb (DHCR7)</h2>
<pre class="r"><code># Show significant association:
data %&gt;% 
  filter(p_SMR &lt; thres &amp; topSNP_chr==11 &amp; topSNP_bp&gt;40000000) %&gt;%
  arrange(Probe_bp)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["probeID"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["ProbeChr"],"name":[2],"type":["int"],"align":["right"]},{"label":["Gene"],"name":[3],"type":["fctr"],"align":["left"]},{"label":["Probe_bp"],"name":[4],"type":["int"],"align":["right"]},{"label":["topSNP"],"name":[5],"type":["fctr"],"align":["left"]},{"label":["topSNP_chr"],"name":[6],"type":["int"],"align":["right"]},{"label":["topSNP_bp"],"name":[7],"type":["int"],"align":["right"]},{"label":["A1"],"name":[8],"type":["fctr"],"align":["left"]},{"label":["A2"],"name":[9],"type":["fctr"],"align":["left"]},{"label":["Freq"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["b_GWAS"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["se_GWAS"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["p_GWAS"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["b_eQTL"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["se_eQTL"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["p_eQTL"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["b_SMR"],"name":[17],"type":["dbl"],"align":["right"]},{"label":["se_SMR"],"name":[18],"type":["dbl"],"align":["right"]},{"label":["p_SMR"],"name":[19],"type":["dbl"],"align":["right"]},{"label":["p_HEIDI"],"name":[20],"type":["dbl"],"align":["right"]},{"label":["nsnp_HEIDI"],"name":[21],"type":["int"],"align":["right"]}],"data":[{"1":"cg09435499","2":"11","3":"FLJ42102*","4":"71117840","5":"rs3763856","6":"11","7":"71125345","8":"A","9":"G","10":"0.663275","11":"0.0243","12":"0.0021","13":"1.449e-30","14":"-0.329494","15":"0.0349091","16":"3.777287e-21","17":"-0.0737495","18":"0.01008330","19":"2.592353e-13","20":"7.498506e-03","21":"19"},{"1":"cg06550274","2":"11","3":"NADSYN1","4":"71150247","5":"rs736894","6":"11","7":"71152258","8":"C","9":"T","10":"0.816826","11":"0.0333","12":"0.0023","13":"3.881e-47","14":"0.878520","15":"0.0413995","16":"6.170747e-100","17":"0.0379047","18":"0.00316934","19":"5.770937e-33","20":"2.865001e-05","21":"15"},{"1":"cg10763288","2":"11","3":"DHCR7*","4":"71158925","5":"rs3794060","6":"11","7":"71187679","8":"T","9":"C","10":"0.785247","11":"0.0352","12":"0.0023","13":"7.743e-54","14":"0.280750","15":"0.0402710","16":"3.135488e-12","17":"0.1253780","18":"0.01976240","19":"2.234549e-10","20":"5.580729e-03","21":"11"},{"1":"cg04774822","2":"11","3":"NADSYN1*","4":"71165839","5":"rs12800438","6":"11","7":"71171003","8":"A","9":"G","10":"0.786702","11":"0.0358","12":"0.0023","13":"1.406e-55","14":"-0.370390","15":"0.0395057","16":"6.877739e-21","17":"-0.0966548","18":"0.01203490","19":"9.652160e-16","20":"6.061979e-03","21":"19"},{"1":"cg27048527","2":"11","3":"NADSYN1*","4":"71184503","5":"rs1792229","6":"11","7":"71179395","8":"A","9":"G","10":"0.825068","11":"0.0350","12":"0.0023","13":"2.905e-51","14":"-0.411415","15":"0.0440426","16":"9.517246e-21","17":"-0.0850723","18":"0.01068610","19":"1.706405e-15","20":"2.032934e-03","21":"12"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="figure">
<img src="IMG/VitaminD_XiaEtAl.rs12785878.png" />

</div>
<pre class="r"><code>tmp &lt;- data %&gt;%
  filter(ProbeChr==11) %&gt;%
  filter(Probe_bp&gt;71000000 &amp; Probe_bp&lt;72000000) %&gt;%
  select(Gene, topSNP, p_SMR, Probe_bp, topSNP_bp) %&gt;%
  gather(key, position, -1, -2, -3) %&gt;%
  mutate(position=position/1000000)
  
tmp %&gt;% ggplot( aes(y=position, x=key)) +
    geom_line(aes(group=paste(Gene,topSNP), color=ifelse(p_SMR&lt;thres,&quot;signif&quot;,&quot;non signif&quot;))) +
    geom_point() +
    geom_text( 
      data=subset(tmp, key==&quot;Probe_bp&quot;) %&gt;% select(Gene, key, position) %&gt;% group_by(Gene) %&gt;% summarize(key=unique(key), position=mean(position)), 
      aes( label=Gene) , angle=45, color=&quot;grey&quot;, hjust=1, nudge_x = -0.1
    ) +
    geom_text_repel( 
      data=subset(tmp, key==&quot;topSNP_bp&quot;) %&gt;% select(topSNP, key, position) %&gt;% group_by(topSNP) %&gt;% summarize(key=unique(key), position=mean(position)), 
      aes( x=key, label=topSNP) , angle=45, color=&quot;grey&quot;, hjust=0, nudge_x = 0.1
    ) +
    coord_flip() +
    theme_ipsum() +
    xlab(&quot;&quot;) +
    ylab(&quot;Position (Mb)&quot;) +
    labs(color=&quot;SMR assoc with Vitamin-D&quot;)</code></pre>
<p><img src="9_Methylation_files/figure-html/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>

&nbsp;
<hr />
<p style="text-align: center;">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>

&nbsp;



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
